django/db/models/expressions.py

```diff
@@ -483,7 +485,14 @@ def __init__(self, value, output_field=None):
         self.value = value
 
     def as_sql(self, compiler, connection):
-        return '%s', [self.value]
+        val = self.value
+        if self._output_field_or_none is not None:
+            from django.db.models.sql.compiler import SQLUpdateCompiler
+            if isinstance(compiler, SQLUpdateCompiler):
```

Comment:
OK - I took a go at fixing the problems here: https://gist.github.com/jarshwah/76dbc87577b7fec05807

There were a tonne of errors on Oracle, mostly due to the fact that oracle stores the empty string as null, and returns the empty string from null. Our docs suggest not using null=True on CharFields, or really anything that derives from CharFields.

The modifications to `empty_strings_allowed` on GenericIPAddress and FilePathField are bugs, and I'm going to open tickets for each of them separately, so no need to include those lines in this PR.